### ISSUE TITLE
Cannot read property 'key' of undefined

### ISSUE THREAD

**Issue by RaulTsc** [sequelize/sequelize#7932]
## What you are doing?

```js
acceptedCandidates: {
      type: Sequelize.ARRAY,
      field: 'accepted_candidates',
      allowNull: false,
    },
```

## What do you expect to happen?
See nice error telling me that I did not put a type for the elements of the array.

## What is actually happening?
Unhandled error cannot read property key of undefined

```
 TypeError: Cannot read property 'key' of undefined

      at Sequelize.Object.<anonymous>.Sequelize.normalizeDataType (node_modules/sequelize/lib/sequelize.js:1366:64)
      at Sequelize.Object.<anonymous>.Sequelize.normalizeAttribute (node_modules/sequelize/lib/sequelize.js:1378:25)
      at Model.<anonymous> (node_modules/sequelize/lib/model.js:84:32)
```

__Dialect: postgres
__Database version: --
__Sequelize version: 3.30.2

**Comment by austinyearlykim** [comment]
it is error at `if (type instanceof DataTypes.ARRAY && dialectTypes[type.type.key]) {`

hmmm.

**Comment by austinyearlykim** [comment]
Got it.

You need to add a type for your array.  What type are the values held by it?

Here is something of the sort below:
`acceptedCandidates: {
      type: Sequelize.ARRAY(Sequelize.TEXT),
      field: 'accepted_candidates',
      allowNull: false,
    },`

yay

**Comment by RaulTsc** [comment]
Makes sense @austindebruyn . Still, it's a weird error message. A nice message would be useful.

**Comment by austindebruyn** [comment]
You most likely meant to tag @austinyearlykim 😃

**Comment by alexanmtz** [comment]
In my case I was using array that is supported just by postgres and in my local env is not supported, so after remove the array datatype it works

**Comment by sushantdhiman** [comment]
Fixed by #9649
